### DISCUSSION TITLE
GTE dataset (with working GTE mode and selectable driving profiles)

### DISCUSSION THREAD

**Discussion by jilleb** [jilleb/MQB-FPA#32] [Modified profiles]
Based on dataset O2 (from Magotan), heavily modified to have GTE mode working again.

[3Q0 - AD - O2 - V03935295AS - Magotan GTE (with FPA) - ultimate working GTEmode.zip](https://github.com/jilleb/MQB-FPA/files/6547519/3Q0.-.AD.-.O2.-.V03935295AS.-.Magotan.GTE.with.FPA.-.ultimate.working.GTEmode.zip)

**Comment by slateksi** [comment]
Hi, Thank you for this! Could you share the full GTE dataset, not just the FPA part? I would flash my gateway to a newer revision and it needs to be re-parametrized.

**Comment by Carmaffia** [comment]
Thanks a lot for sharing!

Did make a 010 conversion to Odis dataset. Dataset loads without any problems onto the 3Q09070530m gateway. Almost happy, because i cannot select, nor have, any driving profile. The buttons are not working, blinks one time yellow and that's it. Even the e-mode menu is gone. So i'm back to my old dataset. Still no GTE function. But someday, somehow it's gonna work.

**Reply by slateksi** [reply]
Yeap had the same with revision H. Needs gw fw revision Q or newer.

**Reply by Carmaffia** [reply]
There's a Q version on my bench. But that's not EV version i think. But see what's happening......

**Reply by jilleb** [reply]
Sometimes you need bus rest before things work on older firmwares.

**Reply by Carmaffia** [reply]
Ah, and how do we force bus reset?

**Reply by jilleb** [reply]
On my car, after a night of rest, I got profile selection. Also, make sure you do a power cycle after flashing the dataset.

**Reply by Carmaffia** [reply]
Bus reset: Open window, remove key and leave car. Lock car. Unlock and you're done.

**Comment by Carmaffia** [comment]
Will do, thanks! Just some rest, we don't have😎

Op wo 26 mei 2021 om 21:15 schreef Chillout ***@***.***>:

**Reply by slateksi** [reply]
Did you get any troublecodes to gw? I got ”data set implausible” with my H revision. Flashed with VCP. Not sure if that will disappear with bus reset? At least it would not clear manually. Maybe I’ll try with odis-e if yours wake up after a nights rest.

**Reply by Carmaffia** [reply]
Flashed with Odis-E, no errors.

**Reply by Carmaffia** [reply]
Did get error while uploading dataset in Q version Gateway. But no problem in M gateway.

**Reply by Carmaffia** [reply]
No succes, after disconnecting battery. Problem stays.

**Reply by slateksi** [reply]
Yeap same here, flashed the modified FPA dataset to H-revision gateway today (with correct start address), buttons not working (e-mode or gte), just flashes quickly when pressed. CAN GW has DTC 00601 Data record implausible. I would flash the gateway to newer FW (I got AA and AC frf:s) but dont have the full GTE dataset for newer revision. 

Flashed the stock H dataset to the gw and e-mode button functions normally again, and GTE button does not. 

Funny thing is that with the Stock pre-fl GW, GTE button works as intended, but e-mode button does not open the menu on screen. :) Its like mix and match.

**Reply by monstermolt** [reply]
Q is not a GTE.
Hybrid is always one letter ahead.
so you need R

**Reply by lito69** [reply]
Can you upload dataset the original H of golf gte? In the gte only the gw that start with 6 work? can the gw dataset that starts with 4 or 5 be flashed in the one that starts with 6?
Thank you

**Comment by Carmaffia** [comment]
Yes, original dataset gives you driving modes but not GTE mode. The VCP
dataset does it exactly reversed. Gives GTE mode but EV modes are not
working. So, who has a mixed dataset?

Op do 27 mei 2021 om 11:31 schreef slateksi ***@***.***>:

**Reply by Carmaffia** [reply]
Somebody got this one working?

**Reply by jilleb** [reply]
I had it running on my GTE for 3 years.

**Reply by Carmaffia** [reply]
Did you also update ECU software, or was it original?

**Reply by jilleb** [reply]
You need to update your ECU if you have installed an AID

**Reply by Carmaffia** [reply]
No AID, so it should be working theoretically. Which it does'nt... I'm out of ideas and options.

**Reply by Leonfr06** [reply]
Any good news??? Someone have gte button works again?

**Reply by Carmaffia** [reply]
No, not on my side. But maybe someone else was succesfull, and can tell us the solution ;-)

**Reply by Leonfr06** [reply]
🤞🏻...,

**Reply by Carmaffia** [reply]
Q/R/M EV gateways are not fully accepting modified dataset. There's a 3q0907530ad (=hard to find) on my bench. It's accepting dataset and seems to be working in the car. To be sure i have to remove CP first. Will report back!

**Reply by Leonfr06** [reply]
Good News!! 🥹

**Reply by Leonfr06** [reply]
More News??

**Reply by Carmaffia** [reply]
I can confirm that this setup works. You need 3q0907530AD gateway, remove CP and transferring SWaP for the existing driving modes. Do SVM on new gateway. After restart you have only 4 driving modes when you hit the GTE button, but all these are electric only. The e-Mode button fails. Upgrade gateway with Jille's magotan dataset and you're ok. 5 driving modes and the last one gives you GTE power. e-Mode button is also functional including hybrid drive. This Saga ends here.....Happy!

**Reply by monstermolt** [reply]
very good. 
With AD Gateway, the door control unit still has to be replaced. 
At least on my mj16. 

I would also have thought there were problems with the large facelift battery. could you put the complete DS online. 
I can't get along with the editor.

**Reply by Carmaffia** [reply]
[19_3Q0907530AD_only_FPA_20220407_151127.zip](https://github.com/jilleb/MQB-FPA/files/8548699/19_3Q0907530AD_only_FPA_20220407_151127.zip)
Here you go!

**Reply by monstermolt** [reply]
Thank you.
I need the complete DS not only FPA.

**Reply by Carmaffia** [reply]
This it all i have, and it's working. Complete dataset you have to source for yourself 🥇

**Reply by monstermolt** [reply]
ok then i'll look for it.

But then I take the AB.

Then I can leave the door control units in.

Or do you not have an error message?

**Reply by Carmaffia** [reply]
I don't have errors in door control units.

**Reply by monstermolt** [reply]
then you don't have side assist.
then there are problems with communication.

**Reply by Carmaffia** [reply]
I don't have side assist, correct.

**Reply by Carmaffia** [reply]
Looking for the OEM 3q0907530AD dataset. Someone?

**Reply by monstermolt** [reply]
yes I'm still looking

**Reply by Carmaffia** [reply]
Only strange thing that happens: My GOM gives a full range (on a fully charged battey) of 32km instead of the old 44km.

**Reply by monstermolt** [reply]
is with me also with R Geteway.
before always 48-52 km now 25-35km.
I think it is calculated more precisely.
If I drive very, very sparingly, I'll get ~47km again.
but I live in the mountains, it's difficult there.

with you comes to that, you have a facelift gateway where the battery is bigger. could also have something to do with it.

**Reply by Carmaffia** [reply]
Good to hear, there's nothing different in coding and adaption to the original (5qe) gateway. I tried the 3q0907530M before and that gives me the same figures as my original GW. 47 km? Never get even close, about 30km in real life, granny style  ;-). I think this one is also in dataset.

**Reply by monstermolt** [reply]
in reality, the 30km fit better. 
gateway AB didn't seem to exist for the Passat GTE, it wasn't built for a while. 

AD I have the SW but not the DS yet. I'm still trying to get it. 
otherwise I have to do it like you. Buy AD from Passat. 

I bought new door control units and 2q Siede assist is also coming.

**Reply by razvanphp** [reply]
Did you guys find the AD full dataset? For me ODIS online removed CP but refuses to parametrise (bad part number for control module), probably because the preFL Golf GTE were never above 5Q...530C gateways.

PS: the original Magotan file posted on top is supposed to work also in Golf GTE? What extension does it have? Tried to flash it with VCP but it's not recognised... wants .zdc 🙁

**Reply by JimVWGTE** [reply]
I also looking for full correct dataset for my Passat GTE with a 3Q...AD

Also have the issue that it shows only 27km on a full charged battery

**Comment by monstermolt** [comment]
I have an 3Q0907530R Gateway in my 2016 GTE without FPA selection.

Although it switches to GTE mode, it doesn't switch back properly.

The driving modes are then mixed up.

A9 sound in all modes.
AID in the Auto Hybrid shows GTE

I got the engine and DSG SW but didn't install it.

**Comment by Libar19** [comment]
Hello everyone, I have a Golf GTE 2017 (Mk7.5) with DCC.
here is my gateway
log part n°: 3Q0 907 530H
Mat: 3Q0 907 530D
Serial number: 27051710520368

I don't have the driving mode (like all GTEs). is it possible to add it? I would like to have it like on the Passat GTE (Mode with Sport, comfort, eco and individual + EV mode

What should be ?

**Comment by RikRocket** [comment]
Okay hopefully anyone here can help me out, because I think I have all the ingredients to the puzzle, but as this is not my cup of tea I could use some guidance.

Upgraded my GTE from mib 1:
New mib installed through Odis, CP removed & all working
Bought a 3Q0907530AD (new)
Installed the gateway, removed fault codes & CP , downloaded the Passat GTE software  (all through Odis) - working

last step is to get jillebs software on the gateway to get GTE button working. My installer is willing to help and tried through ODIS but doesn’t know how to get it on. 

Anyone that can help me by describing what steps he should take in Odis in order to get the modded part from jilleb on there?

thanks so much!

**Reply by Carmaffia** [reply]
Hi Rik,

You need Odis-E to upload dataset into gateway.

Ed

**Reply by Leonfr06** [reply]
Is it possible with vcp??

**Reply by Carmaffia** [reply]
Could be, never tried.

**Reply by mh254** [reply]
> Is it possible with vcp??

Yes, it is.

**Reply by RikRocket** [reply]
Does anyone of you also lives (or knows someone) in the Netherlands, in possession  Odis E or VCP & willing to help me get it on my Gateway? As of right now the guy that I know had no succes in trying, so no GTE mode yet 🥲

Ps. Ed, thx for your recommendation on VW forum, did send him a message, but as he didn’t confirm that he can do it yet, it seemed smart to ask around a bit more.

**Reply by jilleb** [reply]
Me 😁

**Reply by RikRocket** [reply]
![00FD7995-2485-4CF1-AC50-4658292FA14D](https://user-images.githubusercontent.com/120733839/208969020-73c5ccf6-8e70-4ff0-92e3-b848cdc9a274.jpeg)
![EA389C6D-6746-4A06-8C21-996D0A832728](https://user-images.githubusercontent.com/120733839/208969028-08e65c98-837d-436d-a1fd-68dddbec9886.jpeg)
With some help I managed to get it on 😁. Now all that’s left is figuring out how the driving profiles affect the hybrid functions & what my preferences are. 

After monitoring actual consumption data while switching between modes I found out that whenever I switch between driving profiles it automatically puts it in GTE or Battery Hold & the only way to get it back in hybrid auto is , by putting it in Normal drive profile, then put it in Ev mode, then activate hybrid auto.

I’ll fiddle around a bit more in the coming weeks but for now my best guess is that this has to do with the fact that driver profiles overwrite the EV setting in some way. But hey, once you know this it isn’t a problem it’s just 2 presses of a button. 

And thx again to everyone who helped making this possible.  Finally I have a fully functioning MiB2 unit with CarPlay in my MY15 GTE😁

**Reply by Carmaffia** [reply]
Yes, you have to switch back to normal mode, then the e-mode functions correctly. If you don't, you get GTE in display when in full electric mode. It's a small quirk. Enjoy your new set-up!

**Reply by razvanphp** [reply]
@RikRocket so you have a Golf GTE, not Passat? Is the same dataset working on both? Also from Magutan?

And how exactly is this going to be written with VCP? After unzip, the file has no extension and VCP does not recognize it as dataset file (expects .zdc)

Thanks you in advance for shedding some open-source light on this 🙂

**Reply by Carmaffia** [reply]
Goto mqbtools/fpa and convert dataset to VCP format.

**Reply by razvanphp** [reply]
which dataset? The one from this topic? Was confused about what version to use, Magutan 0200 or other Golf GTE versions like Pxxx? Also, how do I know if it's a full dataset or just a diff?

**Reply by Carmaffia** [reply]
Dataset on top of this page. You can upload it. It's only FPA dataset.

**Reply by razvanphp** [reply]
Got it, will try it out, thank you!

BTW, what if I would use a VIN from a golf 7.5 GTE with FPA from factory when removing my fuse from ECU and force ODIS online to give me the AD full dataset? Maybe I can save it from Program Files and share it with everybody.

**Reply by Carmaffia** [reply]
Just try, but also be sure that you have to remove CP again. If you can find full dataset, just upload it right here!

**Reply by Mr-MIBonk** [reply]
Greetings from Carlos...

[19_3Q0907530AD_E_Golf_2019.zip](https://github.com/jilleb/MQB-FPA/files/12535050/19_3Q0907530AD_E_Golf_2019.zip)
[19_3Q0907530AD_V03935313SG PassatB8_GTE.zip](https://github.com/jilleb/MQB-FPA/files/12535053/19_3Q0907530AD_V03935313SG.PassatB8_GTE.zip)

**Reply by razvanphp** [reply]
Wow, thank you Sir! Would they work on Golf GTE too? I guess it's not the same...

**Reply by razvanphp** [reply]
Ok, tried them today, written successfully with VCP after conversion in ODIS-Explorer (nobody mentions this), but with the new 3Q.. 530AD gateway the GTE button stops working, flashes amber and turns off.

So I now suspect following things:
- my AD gateway was brand new, so no FeCs inside, I just managed to do CP online with ODIS, but no FeC and parametrisation.
- MIB2.5 now does not display any driving modes when I press the GTE button
- My old gateway 5Q... 530C still has GTE mode and I can switch to e-mode too, but again, nothing displayed on MIB and I can't select "Charge mode"

So basically the question now is... how can I transfer the SWaP codes from old gateway to the new one?

Also regarding the old one, I've tried to load [those datasets](https://github.com/chobott/MQB-19-Gateway-EV-GTE/tree/main/Datasets), but nothing changed with the old gateway.

**Reply by Carmaffia** [reply]
So CP was succesfully removed, but the rest got f*cked-up. Did you try Odis online and look for faults in gateway? I have the same idea, that FEC/SWaP were not succesfull transferred. That's needed to get the basic driving modes working. The dataset only works if driving modes are ok.

**Reply by razvanphp** [reply]
I've tried with ODIS online, but it does not want to do the Guided function, it says the 3Q AD gateway is wrong part number for my (2015, preFL) car. I've checked in ETKA and indeed, latest gateway for my VIN is 5Q 530C. I've put back my old gateway and this went through, not sure how to pass that.

Or maybe it's the wrong function? Should I use module replacement function or just parametrisation/swap restore?

PS: but given that GTE mode works with my old gateway, shouldn't it display "something" on the MIB2.5 screen?

**Reply by Carmaffia** [reply]
My 5QE gateway did partly function in Mib2.5. Only GTE button was not functional. So you have to check coding/adaption an mib2.5 also. Does he gives FPA functions on display in cockpit?

**Reply by razvanphp** [reply]
Yes, Instrument cluster shows GTE mode, Battery hold and E-Mode.

In the MIB2.5 I've checked all adaptations for `menu_display_charisma`, all are active, not sure what else can be wrong.... donor car was an Arteon.

**Reply by Carmaffia** [reply]
On module 5F:  Coding: 027380010000000071110101009108121F0100C401200D00FE

**Reply by razvanphp** [reply]
thank you @Carmaffia, tried to match your things, but without success. I think you have sound system and I don't, I do have usb hub, you don't, but even excluding those, other changes I did looking at your looking at you coding didn't clear the "Incorrectly coded module" 🙁

**Reply by razvanphp** [reply]
Just for posterity, this was because the new gateway 3Q AD did not have the FeC restored, did that and now modes are back!

Only thing left: MIB2.5 does not display Hybrid menu, it only pops-up when pressing the GTE button, which also does not stay amber anymore. Is this "normal"? Is it there for Mk7.5? I want to mention that I have the e-manager and hybrid display data with energy flow etc.

![172146157_1283386688723219_1070293095652234793_n](https://github.com/jilleb/MQB-FPA/assets/4599319/857c33ca-3af5-4960-a3f1-9d2daf5f7a3d)

**Comment by Mr-MIBonk** [comment]
Good work! Would you like to share the dataset for other people?

**Reply by Carmaffia** [reply]
Dataset is on top of this page. Free!

**Reply by RikRocket** [reply]
Like Carmaffia says, it’s above here. He is the one that helped me get it on 😉

**Reply by lito69** [reply]
That car sounds like someone who spoke to me on telegram

**Reply by gus-kfz** [reply]
Hi, this file in which form? It is not XML and not ZDC, can you share XML one please?

Thx :)

**Reply by gus-kfz** [reply]
Hi, this file in which form? It is not XML and not ZDC, can you share XML one please?

Thx :)

**Reply by Carmaffia** [reply]
Did you unzip first?

**Reply by gus-kfz** [reply]
yes I unzipp it but the file didnt work in ODIS and in VCP

**Comment by RikRocket** [comment]
Hey Razvan,

It’s quite some time since I successfully managed to get this working, with some crucial help from Ed -without him I would probably still be stuck- but yes this it’s a Golf GTE. So it definitely will work. Don’t give up when the attempt fails. Eventually you’ll be able to fix it or find someone near you that can.

Best of luck!

**Comment by RikRocket** [comment]
(To be clear) I mean the dataset above. And the things Ed said. Not the things in the last message in this thread with all those files

**Comment by lito69** [comment]
Can you write if the original gte dataset works? Or maybe the problem is that in 5q0 the fec are in one position and with 3q0 they are in another position?

**Comment by Carmaffia** [comment]
*.xml file works in Odis. Check: http://www.mqbtools.nl/fpa/ for other
options.

Op za 16 mrt 2024 om 15:36 schreef gus-kfz ***@***.***>:

**Reply by gus-kfz** [reply]
in that website http://www.mqbtools.nl/fpa/ I dont find any tips how to get this dataset

**Comment by Carmaffia** [comment]
You can upload it, edit if you want and convert it to VCP/Odis format.

Op za 16 mrt 2024 om 18:18 schreef gus-kfz ***@***.***>:
